tests/components/gardena_bluetooth/test_sensor.py

```diff
@@ -52,3 +52,28 @@ async def test_setup(
         mock_read_char_raw[uuid] = char_raw
         await scan_step()
         assert hass.states.get(entity_id) == snapshot
+
+
+async def test_connected_state(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    mock_entry: MockConfigEntry,
+    mock_read_char_raw: dict[str, bytes],
+    scan_step: Callable[[], Awaitable[None]],
+) -> None:
+    """Verify that a connectivity error makes all entities unavailable."""
+
+    mock_read_char_raw[Sensor.connected_state.uuid] = Sensor.connected_state.encode(
+        False
+    )
+    mock_read_char_raw[Sensor.battery_level.uuid] = Sensor.battery_level.encode(45)
+
+    await setup_entry(hass, mock_entry, [Platform.SENSOR])
+    assert hass.states.get("number.mock_title_sensor_battery_level") == snapshot
```

Comment:
Both snapshots in this function are None, do we really want to use snapshots for that?

Reply:
Good catch. The test was invalid. Snapshots are nice and all.. But they do hide the check somewhat. Would almost be nice if it could update the .py file itself with actual data.